`Hds::Icon` Documentation Updates (#2221)
hashicorp/design-system: website/docs/components/icon/partials/guidelines/overview.md

## What is an icon

> An icon is used to provide visual flourish to text, and in some rare contexts meaning on its own.

Reviewer: @majedelass 👆

> An icon is used to provide visual flourish to text, and in some rare contexts meaning on its own.

Reviewer 2: View here: https://github.com/hashicorp/design-system/pull/2221#discussion_r1691815464

> In the context of the HDS components there is an important distinction to make:

Reviewer 3: ```suggestion
In the context of the HDS components, there is an important distinction to make:
```

> - in Figma the term "icon" can be used interchangeably to indicate the Figma component and the visual element itself (or "glyph")

Reviewer 3: ```suggestion
- In Figma, the term "icon" can be used interchangeably to indicate either the Figma component or the visual element ("glyph") itself.
```

Reviewer: This whole text has been removed now

> - in code the "icon" as glyph is nothing more than an `<svg>` HTML element, while the `Hds::Icon` is the actual Ember component that renders the `<svg>` element.

Reviewer 2: I'm wondering if this is necessary to say (from line 5 -> 8)? Because even in Figma the icon is also an SVG.

Reviewer: I see your point, but it's slightly different, if you think about it: first of all in Figma the glyphs are not `<svg>` but paths (a `.svg` file can be exported, similarly to a `.png`); plus, there is not an `Icon` component, where the content is swapped, but each icon **is** a component (in the [HDS Icons library](https://www.figma.com/design/TLnoT5AYQfy3tZ0H68BgOr/HDS-Icons?t=RMDTpPXI1iCRqwkv-0))

Reviewer 2: That's true, but I'm wondering if saying this will be useful to the consumer. I'll leave others to chime in too.

> In the context of the HDS components there is an important distinction to make:
> 
> - in Figma the term "icon" can be used interchangeably to indicate the Figma component and the visual element itself (or "glyph")
> - in code the "icon" as glyph is nothing more than an `<svg>` HTML element, while the `Hds::Icon` is the actual Ember component that renders the `<svg>` element.

Reviewer: ```suggestion
```

> You can see all the available icons ("glyphs") in [our icon library](/icons/library), also known as Flight Icons library.

Reviewer 3: I don't think we need to keep repeating the alternate "glyph" name. I would just mention it once. I would think the majority of people would use & understand the term "icon" vs. "glyph" anyway.

Reviewer: It's now removed above

## What is an icon

> An icon is used to provide visual flourish to text and, in some rare contexts, meaning on its own.

Reviewer 2: @alex-ju @didoo Maybe this is a better way to phrase this?
@hashicorp/hds-design some input on this would be helpful. 

```suggestion
An icon is used to provide additional meaning to text and, in some rare contexts, meaning on its own.
```

Reviewer 4: +1, icons aren't often used purely for visual flourish.

Reviewer 5: +1 on the flourish language, another note: we explicitly say that icons shouldn't be used on their own/in isolation, but here we're reference the icon having it's own meaning. Is that something that should be excluded or will consumers be confused by this?

## Resources

> - [Icon library](/icons/library)

Reviewer 2: I would create a tab for resources. I can add to the PR if that's what we want?

> - [Icon library](/icons/library)

Reviewer: OK for me, if it work for @zamoore

> - [Icon library](/icons/library)

Author: @majedelass - I reviewed the added Resources tab. Looks good to me. Thanks for adding.
